src/bin/read_csv/mod_dta.c

```diff
@@ -387,39 +386,98 @@ static readstat_value_t value_double_dta(const char *s, size_t len, struct csv_m
 }
 
 static readstat_value_t value_double_date_time_dta(const char *s, size_t len, struct csv_metadata *c) {
-    char *dest;
-    readstat_variable_t *var = &c->variables[c->columns];
-    double val = strtod(s, &dest);
-    if (dest == s) {
-        fprintf(stderr, "not a number: %s\n", (char*)s);
+    // Handle empty or NULL strings as missing values
+    if (s == NULL || len == 0 || *s == '\0') {
+        readstat_value_t value = {
+            .type = READSTAT_TYPE_DOUBLE,
+            .is_system_missing = 1,
+            .v = { .double_value = NAN }
+        };
+        return value;
+    }
+
+    // Truncate the date string to 23 characters to remove the timezone offset and
+    // microseconds, if present. STATA does not support timezones or microseconds.
+    char date_time[24];
+    strncpy(date_time, s, 23);
+    date_time[23] = '\0';
```

Comment:
`snprintf(date_time, sizeof(date_time), "%s", s)` is less error-prone

Reply:
Fixed and tested.